File: src/main/java/com/indiepost/repository/ProfileRepository.java
package com.indiepost.repository;

import com.indiepost.model.Profile;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProfileRepository extends CrudRepository<Profile, Long> {
    List<Profile> findByIdIn(List<Long> ids);

    List<Profile> findAllByOrderByCreatedAtAsc();

    List<Profile> findAllByOrderByCreatedAtDesc();
}

File: src/main/kotlin/com/indiepost/repository/ProfileRepository.kt
package com.indiepost.repository

import com.indiepost.enums.Types
import com.indiepost.model.Profile
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ProfileRepository {

    fun save(profile: Profile): Long?

    fun delete(profile: Profile)

    fun deleteById(id: Long)

    fun findById(id: Long): Profile?

    fun findBySlug(slug: String): Profile?

    fun count(): Long

    fun findAll(pageable: Pageable): Page<Profile>

    fun findAllByProfileType(profileType: Types.ProfileType, pageable: Pageable): Page<Profile>

    fun findAllByFullName(fullName: String, pageable: Pageable): Page<Profile>

    fun findByIdIn(ids: List<Long>, pageable: Pageable): Page<Profile>

    fun findByFullNameIn(profileNames: List<String>, pageable: Pageable): Page<Profile>

    fun countAllByProfileType(profileType: Types.ProfileType): Long

}
